providers/src/airflow/providers/standard/operators/python.py

```diff
@@ -732,12 +709,10 @@ def __init__(
                 f"Sys version: {sys.version_info}. Virtual environment version: {python_version}"
             )
         if python_version is not None and not isinstance(python_version, str):
-            warnings.warn(
-                "Passing non-string types (e.g. int or float) as python_version "
-                "is deprecated. Please use string value instead.",
-                RemovedInAirflow3Warning,
-                stacklevel=2,
+            raise AirflowException(
+                "Passing non-string types (e.g. int or float) as python_version not supported"
```

Comment:
As standard provider is installed as transitive dependency of other providers also in Airflow 2.x... and these are using the new operators... I'd propose we keep the deprecation warning for now. Or we need to conditionally raise if we are running in Airflow 3+ and just emit a warning in Airflow 2?

Reply:
I remember now—there was a discussion about making imports function properly when a user tries to import airflow.operators.python, redirecting it to airflow.providers.standard.operators.python using something like a meta path, when airflow 3.0 released.